docs/proposals/20230123-azure-service-operator.md

```diff
@@ -0,0 +1,232 @@
+
+#### Installation
+
+ASO is a self-contained component published independently of CAPZ. When CAPZ is relying on ASO for critical functionality, CAPZ's installation process should also automatically install ASO.
+
+When installing CAPZ with `clusterctl init`, the Kubernetes resources defined by CAPZ's `infrastructure-components.yaml` published on the corresponding release are installed. ASO can also be [installed with a YAML manifest](https://azure.github.io/azure-service-operator/introduction/installing-from-yaml/) published for each of its releases describing its components that can be incorporated with CAPZ's `infrastructure-components.yaml`. This method installs ASO in the `azureserviceoperator-system` namespace and expects a Secret named `aso-controller-settings` to exist with Service Principal credentials. The easiest and most flexible way to customize that manifest further would likely be to generate a manifest from ASO's Helm chart with any necessary values overridden.
+
+Other than integrating with CAPZ's `infrastructure-components.yaml`, the `capz-controller-manager` container itself could assume responsibility for installing ASO by using the Helm Go library to install the ASO chart when it starts. Installing ASO this way allows CAPZ to hook in additional logic, such as to check if ASO is already installed in the cluster or to verify ASO is healthy after it's installed, which would not be possible if a static manifest was embedded in `infrastructure-components.yaml`.
+
+In terms of specific options, ASO offers both [single- and multi-operator configurations](https://azure.github.io/azure-service-operator/introduction/multitenant-deployment/). The single-operator option is the default, and assumes a single ASO installation managing resources for the whole cluster. The multi-operator method allows several ASO installations to coexist, each monitoring resources in a unique set of namespaces to enable multitenancy. ASO's single-operator configuration also allows multitenancy with more tightly-scoped credentials. Overall, a single ASO installation will likely be the easier path forward since it's consistent with CAPZ's multitenancy model and doesn't involve managing ASO's CRDs and webhooks across multiple installations.
+
+#### Caveats
+
+CAPZ interacts with some Azure services that do not represent infrastructure, and thus cannot be represented in ASO. Resource Health, for example, is "reconciled" by CAPZ currently by getting a resource's health status and reflecting that in the corresponding CAPZ resource, but does not create or update any distinct Azure resources. The new SDK could be used to implement this existing functionality without affecting other service interfaces' use of ASO.
```

Comment:
Question: 
If a resource in Azure can be specified declaratively via a CR for create/delete operations via ASO why can't  it have a resource health field? 
For example I looked at this https://github.com/Azure/azure-service-operator/blob/main/v2/samples/compute/v1beta/v1beta20201201_virtualmachine.yaml one. 
Or is this just a ASO limitation as of now?

Reply:
I don't think there's anything that would make it impossible to add a resource health field. But since most of the types and reconciliation code for ASO are generated automatically and resource health is exposed from a different API from the resource itself, I can see where that might not be practical to add to ASO in the near term.

Reply:
That would be ASO implementation details and may be we need to discuss more about the feasibility. 
But if that is something that is not achieved by ASO then this kind of defeats the purpose because even with ASO capz still needs to use sdk and needs to know how to authenticate to azure cloud API. 
Also it is intuitive to think like this --  someone who is going to create/delete the resources from me should be able to tell the state of those. 
( Just putting the thoughts here, not that this is a big blocker or something. )